ISS/views/forum.py

```diff
@@ -293,8 +321,14 @@ def thanked_posts(request, user_id):
 
 def posts_thanked(request, user_id):
     poster = get_object_or_404(Poster, pk=user_id)
-    posts = (Post.objects.filter(thanks__thanker__id=user_id)
-        .order_by('-created'))
+    excluded_threads = []
+    for thread in Thread.objects.all():
+        if thread.forum.member_view_only and not request.user.is_authenticated:
+            excluded_threads.append(thread)
+    posts = (Post.objects
+             .filter(thanks__thanker__id=user_id)
+             .filter(~Q(thread_id__in=excluded_threads))
```

Comment:
instead of this (and the loop above) you should be able to do:

```
.exclude(thread__forum__member_view_only=True)
```

Generally `.filter(~Q(<some predicate>))` can be replaced with `.exclude(<some_predicate>)` which, while they compile to the same SQL, the latter is considered more idiomatic.

More importantly, doing it this way avoids having to do one query to fetch every thread in the DB and the DB engine can figure out how to efficiently (or as efficiently as possible for it) figure out which threads are or aren't in a members only forum without having to pull the full thread table over the wire, then push back the valid ones.